src/include/homestore/replication/repl_dev.h

```diff
@@ -47,6 +47,20 @@ VENUM(journal_type_t, uint16_t,
       HS_CTRL_REPLACE = 3, // Control message to replace a member
 )
 
+// some events need to be handled by upper layer
+VENUM(event_type_t, uint8_t,
+      RD_FETCH_DATA = 0,      // fetch data
+      RD_NO_SPACE_LEFT = 1,   // no space left error
```

Comment:
Shouldn't we change the signature of `ReplDev::aysnc_alloc_write` to return no space left in return value (since block alloc is being done in sync way) so that HomeObject have a context when no space left is happening. Registering event handler is also okay, but seems unnecessary, IMHO. 

This signature change will also be helpful for NuBlox 2.0.

Reply:
`ReplDev::aysnc_alloc_write` is only called by raft leader, and emergency gc only happens to follower when it gets a no_space_left. If leader gets a no_space_left error from `ReplDev::aysnc_alloc_write` , it will directly return failure to client to let it retry.

so this change is not necessary for nuobject. if it is helpful to you, pls make this change in a seperate PR